src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java

```diff
@@ -39,6 +39,28 @@ public BadArgumentValue(String message) {
         }
     }
 
+    public static class OutOfRangeArgumentValue extends BadArgumentValue {
+        private static final long serialVersionUID = 0L;
+
+        public OutOfRangeArgumentValue(final String arg, final double minValue, final double maxValue, final double value) {
+            super(arg, Double.toString(value), getMessage(minValue, maxValue));
+        }
+
+        private static String getMessage(final double minValue, final double maxValue) {
+            final boolean outMinValue = minValue != Double.NEGATIVE_INFINITY;
+            final boolean outMaxValue = maxValue != Double.POSITIVE_INFINITY;
+            if (outMinValue && outMaxValue) {
+                return String.format("allowed range (%.2f, %.2f).", minValue, maxValue);
```

Comment:
It would also be nice to handle the integer valued case correctly in the output strings (and not print out floating point values for an integer argument).